engine/src/test/java/io/zeebe/engine/processor/workflow/multiinstance/MultiInstanceServiceTaskTest.java

```diff
@@ -0,0 +1,436 @@
+import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import io.zeebe.protocol.record.value.BpmnElementType;
+import io.zeebe.protocol.record.value.WorkflowInstanceRecordValue;
+import io.zeebe.test.util.record.RecordingExporter;
+import io.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class MultiInstanceServiceTaskTest {
+
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+
+  private static final String PROCESS_ID = "process";
+  private static final String ELEMENT_ID = "task";
+  private static final String JOB_TYPE = "test";
+  private static final String INPUT_COLLECTION = "items";
+  private static final String INPUT_ELEMENT = "item";
+
+  private static final BpmnModelInstance WORKFLOW =
+      workflow(b -> b.zeebeInputCollection(INPUT_COLLECTION).zeebeInputElement(INPUT_ELEMENT));
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  private static BpmnModelInstance workflow(
+      final Consumer<MultiInstanceLoopCharacteristicsBuilder> builder) {
+    return Bpmn.createExecutableProcess(PROCESS_ID)
+        .startEvent()
+        .serviceTask(ELEMENT_ID, t -> t.zeebeTaskType(JOB_TYPE).multiInstance(builder))
+        .endEvent()
+        .done();
+  }
+
+  @Test
+  public void shouldCreateOneElementInstanceForEachElement() {
+    // given
+    ENGINE.deployment().withXmlResource(WORKFLOW).deploy();
+
+    // when
+    final long workflowInstanceKey =
+        ENGINE
+            .workflowInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariable(INPUT_COLLECTION, Arrays.asList(10, 20, 30))
+            .create();
+
+    // then
+    final Record<WorkflowInstanceRecordValue> multiInstanceBody =
+        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_ACTIVATED)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .withElementId(ELEMENT_ID)
+            .getFirst();
+
+    Assertions.assertThat(multiInstanceBody.getValue())
+        .hasBpmnElementType(BpmnElementType.SERVICE_TASK)
+        .hasFlowScopeKey(workflowInstanceKey);
+
+    assertThat(
+            RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_ACTIVATED)
+                .withWorkflowInstanceKey(workflowInstanceKey)
+                .withElementId(ELEMENT_ID)
+                .skip(1)
+                .limit(3))
+        .hasSize(3)
+        .extracting(Record::getValue)
+        .extracting(r -> tuple(r.getFlowScopeKey(), r.getBpmnElementType()))
+        .containsOnly(tuple(multiInstanceBody.getKey(), BpmnElementType.SERVICE_TASK));
+  }
+
+  @Test
+  public void shouldCreateOneJobForEachElement() {
```

Comment:
This feels redundant to the normal `ServiceTask` testing; I guess it's because we're modifying the executable model? Hm, something to keep in mind when we refactor it